osu.Game/Screens/OnlinePlay/Multiplayer/MultiplayerMatchSubScreen.cs

```diff
@@ -448,6 +448,13 @@ void endOperation()
 
         private void onRoomUpdated()
         {
+            // may happen if the client is kicked or otherwise removed from the room.
```

Comment:
Test failures are due to this addition. It's concerning that this is required in the first place, ie. why this code is not handling the situation:

https://github.com/ppy/osu/blob/88f70e7f29e39daeab2299dfd8d076407c936bc4/osu.Game/Screens/OnlinePlay/Multiplayer/MultiplayerMatchSubScreen.cs#L282-L288

Needs some more in-depth investigation into exactly how we want to handle room being left, since it's handled in three different ways as of this PR.

Reply:
Also this suggests `Schedule` is not enough to make this `Exit` call safe, which is the only thing guarding the other two calls in this class. They can likely break in the same way, in real world scenarios (ie. if the client was disconnected while at a gameplay screen).__

Reply:
Makes sense that `Schedule` isn't enough. Screens can still do things during the transition.

Reply:
I've guarded against this and centralised the logic. It's not completely covered by tests but in manual testing seems to work as expected.